### ISSUE TITLE
Non-trainable non-differentiable baselines

### ISSUE THREAD

**Issue by lucmos** [PyTorchLightning/pytorch-lightning#2208]
## 🚀 Feature
Add the possibility to implement non-trainable and non-differentiable models.

### Motivation
It is often useful to compare models with simple baselines, those baselines often do not require parameters to optimize and are not differentiable at all. 

However, at the moment it is not possible to return `None` from `training_step` and `configure_optimizers`.
Writing the baselines in the lightning style-guide is useful since it allows to define a parent model that defines how to compute the metrics for all the models.

### Pitch
One common baseline is to _find the most similar sample in the train set and output the associated `y`_. In this way, you can understand if the model is better than simply memorizing the train set.

When comparing models for the same task, it is useful to write the validation step and metrics computations in a parent model and then extend it. At the moment this type baseline is difficult to implement in lightning since it does not support non-trainable models. 

### Alternatives
Define fake parameters, an optimizer to optimize that parameters and a loss to trick the framework into running and computing the validation steps.
